38 ingest portland ridership data into database

jimenasalinas opened the pull request:
Adding script to ingest Portland Ridership data. The script takes the data extracted from the provided TriMet files (JSON format), formats the dates to match model, and then ingests the records into the database. I tried bulk import, but I didn't see a big improvement in speed and there was an issue where Django doesn't support automatically creating incremental ids when using bulk import, so the solution of specifying from which ID number to start felt a bit clunky. Filtering out my JSON file by a subset of dates first did help with the ingestion. 

- I kept my extraction code separate so we could still use Pytest for that script.
- I tried adding duplicate date + id records to the db and the error "Skipping ingestion for station {record['station_id']} due to foreign key constraint violation" seems to catch these instances.
- It would still be good to add some tests once we have the pytest django test db figured out.

JPMartinezClaeys commented on app/scripts/portland_extract.py:
```diff
@@ -0,0 +1,178 @@
+from typing import Dict, Tuple
+
+
+folder_path = ".../CloudStorage/Box-Box/Route Rangers/Transit dataset exploration/Portland Ridership Data/portland_ridership/"
```
This is fine, but I would add this as a global variable if possible.

JPMartinezClaeys commented on app/scripts/portland_ingest.py:
```diff
@@ -0,0 +1,158 @@
+    """
+    Get a list of all id values from the TransitStation table for PDX
+    """
+    return list(TransitStation.objects.filter(city="PDX").values_list("id", flat=True))
```
A query set is already an iterable object so I don't think is necessary to convert to a list.

JPMartinezClaeys commented on app/scripts/portland_ingest.py:
```diff
@@ -0,0 +1,158 @@
+
+                if start_date <= date_obj <= end_date:
+                    # Check if foreign key exists in the list of TransitStation ids
+                    if foreign_key in list_of_db_ids:
```
Given the use that you are given to `list_of_db_ids` I don't think is necessary to create the object as you can check if an object with the id exists by just filtering (as you do on line 94).

Also, since you are retrieving a `TranistStation `object  in line 94, if it exists I don't think it's necessary to check if the foreign key exists in the `list_of_db_ids` as it always should (we are basically retrieving an id from an object from the table and then checking if the id is in the table again). I think this could be adding some inefficiency as it adds one operation on every object inserted.

JPMartinezClaeys reviewed (commented):
There are some comments that I would like if we could address/discuss before approving.

jimenasalinas commented:
changed path into global variable, removed unnecessary filtering and checks, black is activate the checks seem to be pointing to other 3 files in the repo

meganhmoore commented:
> changed path into global variable, removed unnecessary filtering and checks, black is activated in my settings (the checks seem to be pointing to other 3 files in the repo)

Have you pulled in the latest changes from main? Otherwise maybe run `black app` just to make sure everything is cleaned up before going into main

meganhmoore reviewed (approved):
Approved from me if the linking is resolved and tests pass

jimenasalinas commented:
ran black app to format docs causing issues outside of scripts
